pkg/ingester/checkpoint.go

```diff
@@ -239,8 +239,8 @@ func (s *streamIterator) Next() bool {
 	// remove the first stream
 	s.instances[0].streams = s.instances[0].streams[1:]
 
-	stream.chunkMtx.RLock()
-	defer stream.chunkMtx.RUnlock()
+	stream.chunkMtx.Lock()
```

Comment:
toWireChunk writes to `c.blocks[*].offset` on `pkg/chunkenc/memchunk.go:674`
Races with chunk.Bounds(), which doesn't strictly read this value, but does iterate over `c.blocks`.

Reply:
I briefly looked at memchunk.go to see if we could avoid the write to offset, but its not easy to do. Offset is expected to be set after encoding a chunk so all the tests fail if you keep track of offsets separately.